index.js

```diff
@@ -68,6 +88,40 @@ module.exports = {
       srcFile: path.join('config', 'modules.css'),
       destFile: path.join(basePath, 'tailwind.css')
     });
-  }
+  },
+
+  _validateBuildTarget(buildTarget) {
+    if (!buildTarget) {
+      this.ui.writeWarnLine('You must specify a buildTarget using an ember-cli-tailwind config object in your app or addon.')
+      return false;
+    }
+    
+    if (!validBuildTargets.includes(buildTarget)) {
+      this.ui.writeWarnLine('Your buildTarget is invalid. Valid targets are "app", "addon", or "dummy".')
+      return false;
+    }
+
+    if (this._tailwindAddonConfigExists() && !this._isDependency()) {
+      this.ui.writeError('A Tailwind config was detected in the addon folder, but `ember-cli-tailwind` is not listed as a dependency. Please make sure `ember-cli-tailwind` is listed in `dependencies` (NOT `devDependencies`).');
+      return false;
+    }
+    
+    return true;
+  },
 
+  _tailwindAddonConfigExists() {
+    return fs.existsSync(path.join(this.project.root, 'addon', 'tailwind'));
+  },
+
+  _isAddon() {
+    const keywords = this.project.pkg.keywords;
+    return (keywords && keywords.indexOf('ember-addon') !== -1);
+  },
+
+  // Check that `ember-cli-tailwind` is listed in `dependencies` (as opposed to `devDependencies`)
+  _isDependency() {
+    let deps = this.project.pkg.dependencies;
+
+    return Object.keys(deps).includes(this.name);
+  }
```

Comment:
I think all of these checks involving `this.project` will fail for addons that are installed as dependencies rather than being under active development.

```
my-cool-addon
└── ember-cli-addon-docs
    └── ember-cli-tailwind
```

If we have a dependency tree like ☝️, then `this.project` will all be pointed at `my-cool-addon`, but `ember-cli-addon-docs` would be the addon actually trying to set up Tailwind. I think we'll need to be looking at the `includer` argument passed to the `included()` hook rather than `this.project`.

Reply:
@dfreeman good catch! Does that mean we should also change this line in `included` as well?

```
    this.tailwindInputPath = this._getInputPath(this.project.root, buildConfig.path);
```

Or do we actually *want* to be pointing at `my-cool-addon` for this one?

Reply:
I think we only want to be pointing at `my-cool-addon` if we're actually a dependency of `my-cool-addon`, e.g. if the tree looked like this, I'd want the two copies of Tailwind to each only be looking at the addon that's relevant to them:

```
my-cool-addon
├── ember-cli-addon-docs
│   └── ember-cli-tailwind
└── ember-cli-tailwind
```

(In other words, I think we only ever want to use `this.project` for anything if we're dealing with an app, otherwise we'll have different behavior when you're developing an addon vs when you publish it)